From ecefa6c6596982616fa1f564fcee363ef8533808 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Wed, 03 May 2023 12:01:36 +0200
Subject: [PATCH] IWYU: add cstdint for uintptr_t in llvm-10.0 Signals.h

Upstream commit:

  From ff1681ddb303223973653f7f5f3f3435b48a1983 Mon Sep 17 00:00:00 2001
  From: Sergei Trofimovich <slyich@gmail.com>
  Date: Mon, 23 May 2022 08:03:23 +0100
  Subject: [PATCH] [Support] Add missing <cstdint> header to Signals.h

  Without the change llvm build fails on this week's gcc-13 snapshot as:

    [  0%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o
    In file included from llvm/lib/Support/Signals.cpp:14:
    llvm/include/llvm/Support/Signals.h:119:8: error: variable or field 'CleanupOnSignal' declared void
      119 |   void CleanupOnSignal(uintptr_t Context);
          |        ^~~~~~~~~~~~~~~

Change-Id: Iabe0256d86012e7887e295815aa6cfd50694459a
Reviewed-on: https://swiftshader-review.googlesource.com/c/SwiftShader/+/71608
Reviewed-by: Shahbaz Youssefi <syoussefi@google.com>
Kokoro-Result: kokoro <noreply+kokoro@google.com>
Tested-by: Shahbaz Youssefi <syoussefi@google.com>
Commit-Queue: Shahbaz Youssefi <syoussefi@google.com>
---

diff --git a/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h b/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
index e0a18e7..148216b 100644
--- a/third_party/swiftshader/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
+++ b/third_party/swiftshader/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
@@ -14,6 +14,7 @@
 #ifndef LLVM_SUPPORT_SIGNALS_H
 #define LLVM_SUPPORT_SIGNALS_H
 
+#include <cstdint>
 #include <string>
 
 namespace llvm {
